aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -1336,10 +1346,9 @@ impl AptosVM {
                 .map_err(|_| VMStatus::error(StatusCode::STORAGE_ERROR, None))?;
         }
         for (state_key, group_write) in change_set.resource_group_write_set().iter() {
-            for tag in group_write.inner_ops().keys() {
-                // TODO: Typelayout is set to None. Is this correct?
+            for (tag, (_, maybe_layout)) in group_write.inner_ops() {
                 resource_group_view
-                    .get_resource_from_group(state_key, tag, None)
+                    .get_resource_from_group(state_key, tag, maybe_layout.as_deref())
```

Comment:
I think using `None` is ok? This is called foe waypoint, so we do not have any information about groups, so practically  `resource_group_write_set` will be empty.

Reply:
Do we need to explicitly enforce the read-before-write property, in general, or especially for the waypoint, is there anything that explicitly depends on it? (because our executor does not). @igor-aptos @zekun000 @wrwg @runtian-zhou @vgao1996  : if not, can we (or the aptos-vm owner :) delete it, not now, but as a follow up.

Reply:
@gelash @georgemitenkov WriteSetPayload has two variants:
Direct and Script. 

Script is like any other code, and that should be possible to have aggregators and resource groups, and it gets executed here. 

Direct is the one that shouldn't have any aggregators/resource groups in the write set (and that is checked in the execute_write_set function), so resource_group_write_set will be empty, and it will not go here.

so we do need to pass layout here for the "Script" case.